sql: omit NOTICE about indexes not being partitioned for PARTITION ALL BY

otan opened the pull request:
We previously would print a notice specifying that non-partitioned
indexes on partitioned tables may not be performant - however, this
notice does not apply for PARTITION ALL BY tables as they will be
implicitly added for all indexes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59568)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
Nit: "as they will be implicitly added for all indexes." -> "as all of their indexes will be automatically partitioned"

Just nits - <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59568#-:-MSNxAca3IrrUOfjh4kt:b-q3adiy)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/create_index.go, line 422 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59568#-MSNx_7R6Clt3JIIe0ym:-MSNx_7R6Clt3JIIe0yn:bb9g55j) ([raw file](https://github.com/cockroachdb/cockroach/blob/2587ba209fb3d7b00aa8beb674bb7beb1d90001a/pkg/sql/create_index.go#L422)):*
> ```Go
> 	if n.n.PartitionByIndex == nil &&
> 		n.tableDesc.GetPrimaryIndex().GetPartitioning().NumColumns > 0 &&
> 		!n.tableDesc.IsPartitionAllBy() {
> ```

Nit: Can we add a comment here explaining why we exclude IsPartitionAllBy() here.  It makes sense if you're familiar with PARTITION ALL, but if not, a small hint would help readability.
___
*[pkg/sql/create_table.go, line 254 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59568#-MSNxnXg9SyaxQ7LebYa:-MSNxnXg9SyaxQ7LebYb:bkqp0v) ([raw file](https://github.com/cockroachdb/cockroach/blob/2587ba209fb3d7b00aa8beb674bb7beb1d90001a/pkg/sql/create_table.go#L254)):*
> ```Go
> 		for _, def := range n.n.Defs {
> 			if d, ok := def.(*tree.IndexTableDef); ok {
> 				if d.PartitionByIndex == nil && !n.n.PartitionByTable.All {
> ```

Nit: Comment here too.


<!-- Sent from Reviewable.io -->

otan committed:
sql: omit NOTICE about indexes not being partitioned for PARTITION ALL BY

We previously would print a notice specifying that non-partitioned
indexes on partitioned tables may not be performant - however, this
notice does not apply for PARTITION ALL BY tables as all of their
indexes will be automatically partitioned.

Release note: None

otan commented:
thanks

bors r=ajstorm
